scripts/enrich_rulings_ai.py

```
@@ -14,94 +15,136 @@
 
 def ai_extract_provenance_details(
     ruling_text: str, existing_provenance: dict[str, any]
 ) -> dict[str, any]:
     """
-    Simulates extraction of detailed provenance information from ruling text.
-
-    If the ruling text references a Discord ruling, updates the provenance dictionary with a specific source type, source name, and a simulated source date. Returns the updated provenance dictionary.
+    Extracts detailed provenance information from ruling text using an LLM.
+    """
+    logging.info(f"AI: Extracting provenance for: '{ruling_text[:100]}...'")
+    prompt = f"""
```

Ruff (0.11.9)

`info()` call on root logger

(LOG015)